packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_GasPriceOracle.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+
+/* External Imports */
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title OVM_GasPriceOracle
+ * @dev This contract exposes the current execution price, a measure of how congested the network
+ * currently is. This measure is used by the Sequencer to determine what fee to charge for
+ * transactions. When the system is more congested, the execution price will increase and fees
+ * will also increase as a result.
+ *
+ * Compiler used: optimistic-solc
+ * Runtime target: OVM
+ */
+contract OVM_GasPriceOracle is Ownable {
+
+    /*************
+     * Variables *
+     *************/
+
+    // Current execution price
+    uint256 public executionPrice;
+
+    /*************
+     * Constants *
+     *************/
+    uint256 public constant EXECUTION_PRICE_MULTIPLE = 100000001;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _owner Address that will initially own this contract.
+     */
+    constructor(
+        address _owner,
+        uint256 _initialExecutionPrice
+    )
+        Ownable()
+    {
+        setExecutionPrice(_initialExecutionPrice);
+        transferOwnership(_owner);
+    }
+
+
+    /********************
+     * Public Functions *
+     ********************/
+
+    /**
+     * Allows the owner to modify the execution price.
+     * @param _executionPrice New execution price.
+     */
+    function setExecutionPrice(
+        uint256 _executionPrice
+    )
+        public
+        onlyOwner
+    {
+        require(_executionPrice == 1 || _executionPrice % EXECUTION_PRICE_MULTIPLE == 0, "Execution price must be `1` OR a multiple of EXECUTION_PRICE_MULTIPLE.");
```

Comment:
Hmmm according to @tynes this logic is incorrect. I believe we want:

```
        require(
            _executionPrice != 0,
            "OVM_GasPriceOracle: execution price must not be zero"
        );

        require(
            _executionPrice % (10 ** 8) == 1,
            "OVM_GasPriceOracle: execution price must satisfy x % (10**8) == 1"
        );
```

Reply:
Ah yes you're right. Though is the first check redundant?

Reply:
True, pick whichever you want. I feel like 0 as an input is more likely to be a bug that we might want to explicitly surface.

Reply:
Oh whoops sorry only saw your comment after implementing. I just did the 2nd check but I totally hear you about calling out a zero value explicitly. Lemme know if you want me to update it

Reply:
We might as well have both require statements as to prevent bugs

Reply:
We no longer follow this scheme